crates/shared/src/sources/balancer_v2/swap/fixed_point.rs

```diff
@@ -114,6 +114,15 @@ impl Bfp {
         *ONE
     }
 
+    pub fn exp10(exp: i32) -> Self {
+        let exp = exp.saturating_add(18);
+        if exp < 0 {
+            return Self::zero();
```

Comment:
Seems surprising to return `0` here. Is this intended?
If so I think a doc comment about this behavior would be nice so you get informed about it by your IDE when typing this function.

Reply:
I implemented `exp10` to truncate. In particular, we have:

```
Bfp::exp10(-1) == Bfp(1e17)
Bfp::exp10(-2) == Bfp(1e16)
...
Bfp::exp10(-18) == Bfp(1e0) == Bfp(1)
Bfp::exp10(-19) == Bfp(1e-1) == Bfp(0) // truncating
...
```

I will add a doc-comment and some unit tests that demonstrate this.

Reply:
472e804ff0b56c53eca785fe36f67be74362193e